### ISSUE TITLE
lottie: thorvg v1 prerelease rendering regression 

### ISSUE THREAD

**Issue by theashraf** [thorvg/thorvg#3039]
in [dotLottie-rs](https://github.com/LottieFiles/dotlottie-rs), i'm trying to migrate to thorvg v1 and the only change I introduced was modifying the render cycle to clear the buffer before drawing. However, I’m not sure why the animation is getting clipped.

Here’s the updated render cycle code:  
```rs
fn render(&mut self) -> Result<(), LottieRendererError> {
        self.renderer.update().map_err(into_lottie::<R>)?; // calls tvg_canvas_update c api

        #[cfg(feature = "thorvg-v1")]
        self.renderer.clear(false, true).map_err(into_lottie::<R>)?; // calls tvg_canvas_clear c api

        self.renderer.draw().map_err(into_lottie::<R>)?; // calls tvg_canvas_draw c api
        self.renderer.sync().map_err(into_lottie::<R>)?; // calls tvg_canvas_sync c api

        Ok(())
   }
```

Behavior with ThorVG v0.15.6:  
<img src="https://github.com/user-attachments/assets/8b767fd3-f955-4d1c-a00e-08c7efddada1" width="100" />

Behavior with ThorVG v1.0.0-pre7:  
<img src="https://github.com/user-attachments/assets/2bcba599-0933-42c3-920c-0c9f0e8ae305" width="100" />

> This issue is not specific to a certain animation

**Comment by hermet** [comment]
@theashraf could you please share the resource as well?

**Comment by theashraf** [comment]
> @theashraf could you please share the resource as well?

@hermet  [test-lottie.json](https://github.com/user-attachments/files/18089432/test-lottie.json)

**Comment by hermet** [comment]
curprit: https://github.com/thorvg/thorvg/commit/367ac5f331383eea978f07131c64e54fee3f8e58

**Comment by tinyjin** [comment]
Here is difference the commit mentioned and v0.15.x.

The regression issue is come from when use `picture->size` function instead of `scale` after applying mentioned commit.

**[After commit]**
https://github.com/thorvg/thorvg/commit/367ac5f331383eea978f07131c64e54fee3f8e58

When the animation(canvas) size is bigger than given picture size, rendering result is clipped.

![CleanShot 2024-12-27 at 19 14 02@2x](https://github.com/user-attachments/assets/2881bed0-c122-4e36-9a2a-7744d9e96e90)

**[v0.15.x]**

No issue in same condition.

![CleanShot 2024-12-27 at 19 16 26@2x](https://github.com/user-attachments/assets/b6b0baa1-a56f-4117-acff-91d8658c5623)

**Comment by tinyjin** [comment]
Old way of resizing logic was like:

```cpp
// LottieLoader
bool LottieLoader::resize(Paint* paint, float w, float h)
{
    // ...

    //apply the scale to the base clipper
    const Paint* clipper;
    paint->composite(&clipper);

    // ...
}

// tvgPaint
Result Paint::composite(std::unique_ptr<Paint> target, CompositeMethod method) noexcept
{
    // ...

    if (target && method == CompositeMethod::ClipPath) return clip(std::move(target));

    // ...
}
```



The `LottieLoader::resize` function internally calls the `composite()` function. Previously, the old implementation of `composite()` invoked `clip()`, ensuring the clipper was correctly updated to the new size.

However, after the nested clip() call was removed in commit https://github.com/thorvg/thorvg/commit/367ac5f331383eea978f07131c64e54fee3f8e58, the LottieLoader logic was not refactored to accommodate this change. This oversight caused a regression bug where dynamically resized animations failed to update the clipper size correctly.

**Comment by hermet** [comment]
@theashraf the fix will be in v1.0-pre10, thanks.

**Comment by tinyjin** [comment]
![CleanShot 2024-12-30 at 13 40 56@2x](https://github.com/user-attachments/assets/e43e9a34-f0e6-472f-95e7-07b44e1465aa)

Also double-checked, this has been fixed with the `v1.0-pre` + patch cherry-pick in dotlottie-rs
